content/ui-patterns/notification-messaging.mdx

```diff
@@ -0,0 +1,174 @@
+
+Examples:
+- Form field is invalid upon submitting
+- A delete or transfer button
+
+<img
+  width="960"
+  alt="A critical error message on an input field within GitHub Issues."
+  src="https://github.com/primer/design/assets/18661030/5da2e479-6bc7-46dc-9179-46002412b9f5"
+/>
+
+### Neutral
+
+Reserve the `neutral` state for **inline** system related degraded experiences, or for surfacing account related permission messages.
+
+Examples:
+- A row of a data table could not be loaded
+- An item in a list could not be loaded
+- User does not have permissions to perform an action
+
+See [degraded experiences](./degraded-experiences) for more information.
+
+### Upsell
+
+Use the `upsell` state to highlight features that could be unlocked if the user upgrades their account.
+
+Examples:
+- Upgrade account to access a feature
+
+<img
+  width="960"
+  alt="An upsell banner inside the rulesets setting page on GitHub explaing that this feature can be used if the user upgrades to an Enterprise account."
+  src="https://github.com/primer/design/assets/18661030/924884b8-4b85-4967-898d-e0c2fcfaf785"
+/>
+
+## Placement
+
+Message proximity contributes to the prominence of a message. Whenever possible, place messages near the action they are related to.
+
+### Above global navigation
+
+Announcements displayed above the global navigation are reserved for site wide issues that require immediate attention. These messages are not dismissable and should only be used for critical issues that impact the entire site.
+
+<img
+  width="960"
+  alt="Highlights a global announcement banner above the GitHub global navigation bar inside a browser window."
+  src="https://github.com/primer/design/assets/18661030/3d843939-5100-424e-a9a4-d663c517432f"
+/>
+
+### Top of body
+
+If messaging pertains to an entire page or section, place it at the top of the body content. This is the most common placement for messaging components. Banners should fill the container width but maintain appropriate spacing between elements and not appear full width or flush under the global nav.
+
+<img
+  width="960"
+  alt="Highlights a banner at the top of the body inside a GitHub issue."
+  src="https://github.com/primer/design/assets/18661030/73a2ace1-e4fe-4255-850d-062c8d75b7a9"
+/>
+
+### Inline
+
+Inline messages are used to provide feedback on a specific action or input field. They should be placed near the action they are related to.
+
+<img
+  width="960"
+  alt="Highlights an inline message next to an input field inside a GitHub issue."
+  src="https://github.com/primer/design/assets/18661030/ea98bf69-1a7a-4061-8fce-2e2be4303b55"
+/>
+
+### Inside a dialog
+
+Banners should be placed below the Dialog header and appear full width within the Dialog content area. Prioritize showing messaging _inside_ the Dialog after an action rather than closing the Dialog and showing a banner within the page.
```

Comment:
"Prioritize showing messaging _inside_ the Dialog after an action rather than closing the Dialog and showing a banner within the page."

✨yay✨